### PR TITLE
Use Javabeans In Getters and Setters

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public ConfigurationSetting addSetting(ConfigurationSetting setting) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public ConfigurationSetting setSetting(String key, String value) {
-        return setSetting(new ConfigurationSetting().key(key).value(value), Context.NONE).value();
+        return setSetting(new ConfigurationSetting().key(key).value(value), Context.NONE).getValue();
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Shouldn't the `key()` and `value()` method also be renamed? Same in async client too.
```
        return setSetting(new ConfigurationSetting().setKey(key).setValue(value), Context.NONE).getValue();
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 2]
```diff
@@ -225,7 +225,7 @@ public ConfigurationSetting setSetting(ConfigurationSetting setting) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public ConfigurationSetting updateSetting(String key, String value) {
-        return updateSetting(new ConfigurationSetting().key(key).value(value), Context.NONE).value();
+        return updateSetting(new ConfigurationSetting().key(key).value(value), Context.NONE).getValue();
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: same here

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/policy/ConfigurationCredentialsPolicy.java [hunk 3]
```diff
@@ -48,17 +48,17 @@ public ConfigurationCredentialsPolicy(ConfigurationClientCredentials credentials
      */
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        final Flux<ByteBuffer> contents = context.httpRequest().body() == null
+        final Flux<ByteBuffer> contents = context.getHttpRequest().body() == null
             ? Flux.just(getEmptyBuffer())
-            : context.httpRequest().body();
+            : context.getHttpRequest().body();
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/policy/ConfigurationCredentialsPolicy.java]: `getBody()`?

File: sdk/appconfiguration/azure-data-appconfiguration/src/samples/java/com/azure/data/appconfiguration/ConfigurationClientJavaDocCodeSnippets.java [hunk 4]
```diff
@@ -158,13 +158,13 @@ public void setSetting() {
         Response<ConfigurationSetting> responseSetting = configurationClient
             .setSettingWithResponse(new ConfigurationSetting().key("prodDBConnection").label("westUS")
                     .value("db_connection"), new Context(key2, value2));
-        System.out.printf("Key: %s, Value: %s", responseSetting.value().key(), responseSetting.value().value());
+        System.out.printf("Key: %s, Value: %s", responseSetting.getValue().key(), responseSetting.getValue().value());
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/samples/java/com/azure/data/appconfiguration/ConfigurationClientJavaDocCodeSnippets.java]: `getKey()` and `getValue()`

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java [hunk 5]
```diff
@@ -467,7 +467,7 @@ static void assertConfigurationEquals(ConfigurationSetting expected, Response<Co
         assertNotNull(response);
         assertEquals(expectedStatusCode, response.statusCode());
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: `getStatusCode()`

**Comment by alzimmermsft** [comment]
> I started finding methods that weren't renamed and there's going to be quite of few of them. I guess we should use this PR to get as many methods renamed as possible and get it merged.
> 
> Then we can sweep through it again to rename the remaining. This can be a group exercise where each of us would skim through a few assigned packages and update.

I forgot to mention this is only changing Azure Core classes.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java (hunk 1 above)
Comment (JonathanGiles) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Yep, once app configuration is refactored.

File: sdk/core/azure-core-amqp/pom.xml [hunk 6]
```diff
@@ -80,6 +80,7 @@
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
+      <version>${mockito-core.version}</version>
```
Comment (conniey) [inline sdk/core/azure-core-amqp/pom.xml]: I thought thim the parent pom.xmls would be inherited from

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java [hunk 7]
```diff
@@ -317,25 +317,25 @@ private static String getExpression(EventPosition eventPosition) {
...
         }
 
-        if (eventPosition.enqueuedDateTime() != null) {
+        if (eventPosition.getEnqueuedDateTime() != null) {
             String ms;
             try {
-                ms = Long.toString(eventPosition.enqueuedDateTime().toEpochMilli());
+                ms = Long.toString(eventPosition.getEnqueuedDateTime().toEpochMilli());
             } catch (ArithmeticException ex) {
                 ms = Long.toString(Long.MAX_VALUE);
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java]: @srnagar `String eventHubName()` is not renamed.

Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java]: Updated

File: sdk/core/azure-core-amqp/pom.xml (hunk 6 above)
Comment (srnagar) [reply sdk/core/azure-core-amqp/pom.xml]: Removed

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml [hunk 8]
```diff
@@ -382,8 +382,8 @@ page at http://checkstyle.sourceforge.net/config.html -->
     <!-- Fluent method checks:
     (1) A method returns an instance of the class, and that have one parameter,
     (2) should not start with the words in the avoidStartWords list.-->
-    <module name="com.azure.tools.checkstyle.checks.FluentMethodNameCheck">
-      <property name="avoidStartWords" value="with,set"/>
-    </module>
+<!--    <module name="com.azure.tools.checkstyle.checks.FluentMethodNameCheck">-->
```
Comment (sima-zhu) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml]: Why the checkstyle comment out?

**Pull request merged**